### PR TITLE
feat: All releases downloads

### LINKED ISSUE #51: [DESIGN BUG] center the text and use the small text size
Screen Shot 2022-11-28 02 06 44 PM

### PR DESCRIPTION
**Pull request by nhsz**
This PR adds the logic to fetch, parse, process and display the releases data.

## Description

- updates `/pages/downloads.tsx` with the logic to fetch, process and display the releases data
- updates `DownloadsTable` and `DataTable` components to be able to use the real data
- adds `fast-xml-parser` dep to parse XML data
- adds `netlify.toml` config to [support ISR](https://docs.netlify.com/integrations/frameworks/next-js/incremental-static-regeneration/)
- adds `types.ts`
- adds `/utils`: 
  - `compareReleasesFn`
  - `fetchLatestReleaseCommit`
  - `fetchLatestReleaseVersionAndName`
  - `fetchXMLData`
  - `getChecksum`
  - `getLatestBinaryURL`
  - `getParsedDate`
  - `getReleaseArch`
  - `getReleaseCommitHash`
  - `getReleaseCommitURL`
  - `getReleaseKind`
  - `getReleaseName`
  - `getReleaseSize`
  - `getReleaseURL`
  - `getSignatureURL`
  - `getSortedReleases`
  - `mapReleasesData`
- updates `constants.ts`
- removes `/data/test/download-testdata.ts`
- fixes #51

### CODE REVIEW

File: src/components/UI/DataTable.tsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
-import { Table, Thead, Tr, Th, TableContainer, Text, Tbody, Td } from '@chakra-ui/react';
+import { Link, Table, Thead, Tr, Th, TableContainer, Text, Tbody, Td } from '@chakra-ui/react';
 import { FC } from 'react';
+import { ReleaseData } from '../../types';
+import { getParsedDate } from '../../utils';
 
 interface Props {
   columnHeaders: string[];
+  // TODO: update data type
   data: any;
```
Comment (corwintines) [inline src/components/UI/DataTable.tsx:9]:
```suggestion
  data: ReleaseData[];
```

Just looking through code, I think we can add this typing for the TODO

**Review by corwintines — CHANGES_REQUESTED** [review]
Looks good other than the above comment on type. Nice work!

File: src/components/UI/DataTable.tsx (hunk 1 above)
Comment (nhsz) [reply src/components/UI/DataTable.tsx:9]: @corwintines not possible for now as `DataTable` is also being used for OpenPGP Signatures table, where data type is different

Comment (corwintines) [reply src/components/UI/DataTable.tsx:9]: Sounds good, will approve then

**Review by corwintines — APPROVED** [review]
Nice!

**Review by wackerow — APPROVED** [review]

**Comment by wackerow** [comment]
Looks great! Nice job @nhsz! :fire:

**Pull request merged**